# ER Wait Times by Severity Level 
Calculating average ER wait times by severity level. Validating SQL findings with Pandas for accuracy. 

In [1]:
import pandas as pd

## Load ER Visits Data

In [2]:
er_visits = pd.read_csv("../data/raw/er_visits.csv") 

## Preview of ER Visits DataFrame

In [3]:
er_visits.head()

,visit_id,patient_id,arrival_date,arrival_day,arrival_hour,severity_level,wait_time_minutes,length_of_stay_hours,admitted
0,1,329,11/30/2025,Sunday,19,Low,137,4.7,No
1,2,152,1/7/2025,Tuesday,2,High,57,11.6,Yes
2,3,10,11/2/2025,Sunday,5,High,36,8.8,Yes
3,4,283,8/17/2025,Sunday,14,High,10,12.9,No
4,5,333,2/19/2026,Thursday,22,Medium,56,11.0,Yes


## SQL to Pandas Validation: Average ER Wait Times by Severity Level 

In [4]:
er_visits.groupby("severity_level")["wait_time_minutes"].mean().sort_values(ascending=False)

severity_level
Low         108.846958
Medium       73.160622
High         35.224299
Critical      7.405405
Name: wait_time_minutes, dtype: float64

## Breakdown of Validation Results
The higher the severity level of the appoinment, the shorter the overall wait time. Low Severity pts are seen with avg wait time of 109 min
Med Severity are pts seen with avg wait time of 73 min, High Severity pts are seen with avg wait time of 35 min, and Critical pts seen with avg wait time of 7 min. 